server/services/store/sqlstore/blocks.go

```diff
@@ -788,3 +802,264 @@ func (s *SQLStore) duplicateBlock(db sq.BaseRunner, boardID string, blockID stri
 	}
 	return allBlocks, nil
 }
+
+func (s *SQLStore) deleteBlockChildren(db sq.BaseRunner, boardID string, parentID string, modifiedBy string) error {
+	now := utils.GetMillis()
+
+	selectQuery := s.getQueryBuilder(db).
+		Select(
+			"board_id",
+			"id",
+			"parent_id",
+			s.escapeField("schema"),
+			"type",
+			"title",
+			"fields",
+			"'"+modifiedBy+"'",
+			"create_at",
+			s.castInt(now, "update_at"),
+			s.castInt(now, "delete_at"),
+			"created_by",
+		).
+		From(s.tablePrefix + "blocks").
+		Where(sq.Eq{"board_id": boardID})
+
+	if parentID != "" {
+		selectQuery = selectQuery.Where(sq.Eq{"parent_id": parentID})
+	}
+
+	insertQuery := s.getQueryBuilder(db).
+		Insert(s.tablePrefix+"blocks_history").
+		Columns(
+			"board_id",
+			"id",
+			"parent_id",
+			s.escapeField("schema"),
+			"type",
+			"title",
+			"fields",
+			"modified_by",
+			"create_at",
+			"update_at",
+			"delete_at",
+			"created_by",
+		).Select(selectQuery)
+
+	if _, err := insertQuery.Exec(); err != nil {
+		return err
+	}
+
+	deleteQuery := s.getQueryBuilder(db).
+		Delete(s.tablePrefix + "blocks").
+		Where(sq.Eq{"board_id": boardID})
+
+	if parentID != "" {
+		deleteQuery = deleteQuery.Where(sq.Eq{"parent_id": parentID})
+	}
+
+	if _, err := deleteQuery.Exec(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *SQLStore) undeleteBlockChildren(db sq.BaseRunner, boardID string, parentID string, modifiedBy string) error {
+	if boardID == "" {
+		return ErrEmptyBoardID{}
+	}
+
+	where := fmt.Sprintf("board_id='%s'", boardID)
+	if parentID != "" {
+		where += fmt.Sprintf(" AND parent_id='%s'", parentID)
+	}
+
+	/*
```

Comment:
Nit: This comments are going to end up being misleading if we don't update them properly, I prefer to avoid them.